src/rust/ide/src/controller/searcher.rs

```diff
@@ -578,57 +587,150 @@ impl Searcher {
+            let requests  = return_types.into_iter().map(|return_type| {
+                info!(this.logger, "Requesting suggestions for returnType {return_type:?}");
+                let file = graph.module.path().file_path();
+                ls.completion(file,&position,&this_type,&return_type,&tags)
+            });
+            let responses = futures::future::join_all(requests).await;
+            info!(this.logger,"Received suggestions from Language Server.");
+            let new_suggestions = match this.suggestions_from_responses(responses) {
+                Ok(list)   => Suggestions::Loaded {list:Rc::new(list)},
                 Err(error) => Suggestions::Error(Rc::new(error.into()))
             };
-            data.borrow_mut().suggestions = new_suggestions;
-            notifier.publish(Notification::NewSuggestionList).await;
+            this.data.borrow_mut().suggestions = new_suggestions;
+            this.notifier.publish(Notification::NewSuggestionList).await;
         });
-        Ok(())
+    }
+
+    /// Process multiple completion responses from the engine into a single list of suggestion.
+    fn suggestions_from_responses
+    (&self, responses:Vec<json_rpc::Result<language_server::response::Completion>>)
+    -> FallibleResult<Vec<Suggestion>> {
+        let mut suggestions = Vec::new();
+        for response in responses {
+            let response = response?;
+            let entries  = response.results.iter().filter_map(|id| {
+                self.database.get(*id).map_none(||
+                    error!(self.logger,"Missing entry {id} in Suggestion Database.")
+                ).map(Suggestion::Completion)
+            });
+            suggestions.extend(entries);
+        }
+        Ok(suggestions)
+    }
+
+    fn possible_function_calls(&self) -> Vec<CompletionSuggestion> {
+        let opt_result = || {
+            let call_ast = self.data.borrow().input.expression.as_ref()?.func.clone_ref();
+            let call     = SimpleFunctionCall::try_new(&call_ast)?;
+            if let Some(module) = self.module_method_called(&call) {
+                let entry = self.database.lookup_by_name_and_module(call.function_name,&module);
+                Some(entry.into_iter().collect())
+            } else {
+                let name     = &call.function_name;
+                let location = *self.position_in_code;
+                Some(self.database.lookup_by_name_and_location(name,location))
+            }
+        };
+        opt_result().unwrap_or(Vec::new())
+    }
+
+    fn module_method_called(&self, call:&SimpleFunctionCall) -> Option<QualifiedName> {
+        let module_ast      = self.graph.graph().module.ast();
+        let position        = *self.position_in_code;
+        let module          = double_representation::module::Info {ast:module_ast};
+        let this_name       = ast::identifier::name(call.this_argument.as_ref()?)?;
```

Comment:
To be considered what happens if "this_argument" is a qualified name, e.g. we have "Foo.Bar.method". Is this even legal?

Reply:
Actually the only place in enso docs where I found qualified names is only import syntax. So as I understood, you cannot use qualified names in code, only in imports. However this may change in future.